programs/pkey/pk_decrypt.c

```diff
@@ -69,7 +68,10 @@ int main( int argc, char *argv[] )
     const char *pers = "mbedtls_pk_decrypt";
     ((void) argv);
 
+    mbedtls_pk_init( &pk );
+    mbedtls_entropy_init( &entropy );
     mbedtls_ctr_drbg_init( &ctr_drbg );
```

Comment:
Minor: Why initialise these before you even know that the program arguments are correct? same for the other program.

Reply:
@andresag01 Because as it stands, the argument check jumps to the `exit` label, which frees `pk, entropy, ctr_drbg`. But now that you state it, I think we usually make the exception that we perform the argument checks before doing any initialization, allowing for immediate return from that check. I'll fix that along the way in this PR.

Reply:
in this case, we should not `goto exit` if the program arguments are incorrect

Reply:
@andresag01 On windows, these example programs wait for input before returning, and an immediate `return` when noting a bad number of arguments would change that behavior. I therefore prefer to keep it as is.